compiler/rustc_parse/src/parser/path.rs

```diff
@@ -880,6 +880,15 @@ impl<'a> Parser<'a> {
         &mut self,
         ty_generics: Option<&Generics>,
     ) -> PResult<'a, Option<GenericArg>> {
+        let mut attr_span: Option<Span> = None;
+        if self.token == token::Pound && self.look_ahead(1, |t| *t == token::OpenBracket) {
+            // Parse attribute.
+            let attrs_wrapper = self.parse_outer_attributes()?;
+            if !attrs_wrapper.is_empty() {
```

Comment:
Isn't this always true since we *know* from the check above that there's a `#` followed by a `[`?